gateway/internal/controllers/subscription_controller.go

```diff
@@ -0,0 +1,78 @@
+package controllers
+
+import (
+	"context"
+	"errors"
+	"net/http"
+
+	"github.com/GenesisEducationKyiv/software-engineering-school-4-0-boredgus/gateway/internal/clients/dispatch"
+)
+
+type (
+	DispatchService interface {
+		SubscribeForDispatch(ctx context.Context, email, dispatchId string) error
+		UnsubscribeFromDispatch(ctx context.Context, email, dispatchId string) error
+	}
+
+	params struct {
```

Comment:
I guess better naming for this is like 'subscriptionRequest'

Reply:
Renamed due to recommendation